Replace surface

oyvindeide opened the pull request:
Closes #213

lars-petter-hauge commented on semeio/jobs/overburden_timeshift/ots.py:
```diff
@@ -100,18 +102,35 @@ def ots_run(parameter_file):
-                f.write(line.format(xy[0], xy[1], surface_horizon[point], *ts))
+
+            point = 0
+            for x_index in range(1, surface_horizon.get_nx() + 1):
```
We tend to do a lot of these double loops, which is a very natural thing as we are working with grids very often (or even triple for loops, when going through 3 dimensions).

I've always disliked that it is necessary to add more indents than a single one.. 

What do you think about:

`for i, j in itertools.product(range(1, nx+1), range(1, ny+1)):`

A bit annoying that it isn't 0-indexed, but alright..

If you need the `point` as well:

`for point, (i, j) in enumerate(itertools.product(range(1, nx+1), range(1, ny+1))):`

oyvindeide replied:
Yes, will probably do something like this, or perhaps suggest a patch upstream? I think it makes sense for a surface to be iterable, and this is at least what I would expect to get if I iterated over it (it is not iterable at the moment).

oyvindeide replied:
My first goal was just to replace it, then clean up the implementation ;)

xjules commented on semeio/jobs/overburden_timeshift/ots.py:
```diff
@@ -49,10 +48,13 @@ def ots_load_params(input_file):
         d0 = vp[0].strftime("%Y_%m_%d")
         d1 = vp[1].strftime("%Y_%m_%d")
-        ts[iv].write(output_dir + type_str + "/ots" + "_" + d0 + "_" + d1 + ".irap")
+        ts[iv].to_file(os.path.join(output_dir, "ots" + "_" + d0 + "_" + d1 + ".irap"))
```
Should the extension be the same for xtgeo surface?

oyvindeide replied:
It is still `irap`, so I suppose so, but this implementation is failing for one of the tests, so might not be feasible to replace.

lars-petter-hauge commented on semeio/jobs/overburden_timeshift/ots.py:
```diff
@@ -172,34 +190,44 @@ def _create_surface(self, z=None):
-        for i in range(len(surf)):
-            irap_x[i], irap_y[i] = surf.getXY(i)
+        counter = 0
+        for x_index in range(1, nx + 1):
```
Could technically do the same here as above: `for point, (x_index, y_index) in enumerate(product(`

jondequinor commented on semeio/jobs/overburden_timeshift/ots.py:
```diff
@@ -50,10 +50,13 @@ def ots_load_params(input_file):
 
 
 def write_surface(vintage_pairs, ts, output_dir, type_str):
```
Pathlib??

Shouldn't we use pathlib for stuff like this?

```suggestion
def write_surface(vintage_pairs, ts, output_dir, type_str):
    output_path = Pathlib(output_dir + type_str)
    if not output_path.is_dir():
        output_path.mkdir()
    …
        ts[iv].to_file((output_dir / f"ots_{d0}_{d1}.irap"))    
```

jondequinor replied:
🙇

jondequinor commented:
Both the description and issue is a bit scarce, so someone with more knowledge will have to review. Except for the one comment, code looks OK.

oyvindeide commented:
> Both the description and issue is a bit scarce, so someone with more knowledge will have to review. Except for the one comment, code looks OK.

I have added some context in the issue and changed the commit text.

jondequinor reviewed (approved):
Code looks good to me.
